<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>robotWindow</class>
 <widget class="QWidget" name="robotWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1300</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create Robot</string>
  </property>
  <widget class="robotOpenGlWindow" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>850</width>
     <height>700</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>400</y>
     <width>90</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Distance :</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_1">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>50</y>
     <width>161</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Robot File Name :</string>
   </property>
  </widget>
  <widget class="QPushButton" name="newRobotButton">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>10</y>
     <width>220</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>New Robot</string>
   </property>
  </widget>
  <widget class="QPushButton" name="deleteButton">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>997</x>
     <y>300</y>
     <width>106</width>
     <height>28</height>
    </rect>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
  </widget>
  <widget class="QPushButton" name="editButton">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>300</y>
     <width>106</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Edit</string>
   </property>
  </widget>
  <widget class="QPushButton" name="saveDimentionButton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1150</x>
     <y>400</y>
     <width>140</width>
     <height>110</height>
    </rect>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QPushButton" name="addMotorButton">
   <property name="geometry">
    <rect>
     <x>1110</x>
     <y>140</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Motors</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="countspinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1030</x>
     <y>430</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>11</number>
   </property>
   <property name="singleStep">
    <number>2</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="distancespinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>430</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="minimum">
    <number>20</number>
   </property>
   <property name="maximum">
    <number>60</number>
   </property>
   <property name="singleStep">
    <number>5</number>
   </property>
   <property name="value">
    <number>20</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>1030</x>
     <y>400</y>
     <width>60</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Count :</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>340</y>
     <width>410</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>80</y>
     <width>581</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>..</string>
   </property>
  </widget>
  <widget class="QPushButton" name="loadRobotButton">
   <property name="geometry">
    <rect>
     <x>1110</x>
     <y>10</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Load Robot</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>110</y>
     <width>201</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Parts of Robot :</string>
   </property>
  </widget>
  <widget class="QPushButton" name="addLineSensorButton">
   <property name="geometry">
    <rect>
     <x>1110</x>
     <y>170</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Line Sensors</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>1110</x>
     <y>110</y>
     <width>70</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Add :</string>
   </property>
  </widget>
  <widget class="QPushButton" name="addIRPButton">
   <property name="geometry">
    <rect>
     <x>1110</x>
     <y>200</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>IR Proximity Sensors</string>
   </property>
  </widget>
  <widget class="QPushButton" name="addIRDButton">
   <property name="geometry">
    <rect>
     <x>1110</x>
     <y>230</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>IR Distance Sensors</string>
   </property>
  </widget>
  <widget class="QPushButton" name="addSonarSensorButton">
   <property name="geometry">
    <rect>
     <x>1110</x>
     <y>260</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Sonar Sensors</string>
   </property>
  </widget>
  <widget class="QPushButton" name="exitButton">
   <property name="geometry">
    <rect>
     <x>1180</x>
     <y>680</y>
     <width>104</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cancelButton">
   <property name="geometry">
    <rect>
     <x>1060</x>
     <y>680</y>
     <width>104</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Save as</string>
   </property>
  </widget>
  <widget class="QPushButton" name="saveButton">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>940</x>
     <y>680</y>
     <width>104</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QListWidget" name="listWidget">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>142</y>
     <width>222</width>
     <height>140</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="sizeResetButton">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1030</x>
     <y>480</y>
     <width>110</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
  </widget>
  <widget class="QPushButton" name="sizeUndoButton">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>480</y>
     <width>110</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>robotOpenGlWindow</class>
   <extends>QWidget</extends>
   <header>robotopenglwindow.h</header>
   <container>1</container>
   <slots>
    <slot>updateRobotObject(vector_robotObjectType)</slot>
    <slot>currentObject(QString)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>newRobotButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>newRobotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>964</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>876</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadRobotButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>loadRobotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1147</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>1159</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addMotorButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>addMotorButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1192</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>1106</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addLineSensorButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>addLineSensorButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1157</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>1260</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addIRPButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>addIRPSensorButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1204</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>1203</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addIRDButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>addIRDSensorButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1129</x>
     <y>243</y>
    </hint>
    <hint type="destinationlabel">
     <x>1221</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addSonarSensorButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>addSonarSensorButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1180</x>
     <y>282</y>
    </hint>
    <hint type="destinationlabel">
     <x>1232</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>editButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>935</x>
     <y>315</y>
    </hint>
    <hint type="destinationlabel">
     <x>893</x>
     <y>339</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>deleteButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1023</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>1007</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveDimentionButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>saveButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1173</x>
     <y>426</y>
    </hint>
    <hint type="destinationlabel">
     <x>1155</x>
     <y>390</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>saveRobotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1025</x>
     <y>689</y>
    </hint>
    <hint type="destinationlabel">
     <x>1022</x>
     <y>660</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>cancelButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1109</x>
     <y>695</y>
    </hint>
    <hint type="destinationlabel">
     <x>1099</x>
     <y>650</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exitButton</sender>
   <signal>clicked()</signal>
   <receiver>robotWindow</receiver>
   <slot>exitButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1221</x>
     <y>695</y>
    </hint>
    <hint type="destinationlabel">
     <x>1204</x>
     <y>627</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>robotWindow</sender>
   <signal>robotObjectUpdated(vector_robotObjectType)</signal>
   <receiver>widget</receiver>
   <slot>updateRobotObject(vector_robotObjectType)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>922</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>786</x>
     <y>563</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listWidget</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>widget</receiver>
   <slot>currentObject(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>930</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>710</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>distancespinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>robotWindow</receiver>
   <slot>distanceSpinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>921</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel">
     <x>921</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>countspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>robotWindow</receiver>
   <slot>countSpinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1049</x>
     <y>442</y>
    </hint>
    <hint type="destinationlabel">
     <x>1062</x>
     <y>474</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>robotObjectUpdated(vector_robotObjectType)</signal>
  <slot>newRobotButtonClicked()</slot>
  <slot>loadRobotButtonClicked()</slot>
  <slot>addMotorButtonClicked()</slot>
  <slot>addLineSensorButtonClicked()</slot>
  <slot>addSonarSensorButtonClicked()</slot>
  <slot>addIRPSensorButtonClicked()</slot>
  <slot>addIRDSensorButtonClicked()</slot>
  <slot>editButtonClicked()</slot>
  <slot>deleteButtonClicked()</slot>
  <slot>saveRobotButtonClicked()</slot>
  <slot>cancelButtonClicked()</slot>
  <slot>exitButtonClicked()</slot>
  <slot>saveButtonClicked()</slot>
  <slot>currentRowChanged(int)</slot>
  <slot>distanceSpinBoxValueChanged(int)</slot>
  <slot>countSpinBoxValueChanged(int)</slot>
 </slots>
</ui>
